### ISSUE TITLE
Tetra items made with Undergarden mats cause crash.

### ISSUE THREAD

**Issue by Darkest-Traveler** [Syrikal/Tetra-Undergarden-Patch#2]
Tetra items made with undergarden mats can sometimes cause a stack overflow. Its not consistent or repeatable but it only happens with the undergarden mats when it does. here is the last crash log related to it I had. Thank you for your time.

https://pastebin.com/mNVFciWC

**Comment by Darkest-Traveler** [comment]
here is another crash log related to it player was upgrading the forgotten sword and was upgrading the counterbalance with utherium.

https://pastebin.com/NbTgYwfa

**Comment by Syrikal** [comment]
I was going to say that I've done some testing and haven't been able to replicate the bug, but then I saw your new report. I have a couple questions.

- Does this only happen with Forgotten Metal?
- When exactly does the crash occur?

If you answer those I _might_ be able to fix it, but without being able to replicate it it'll be pretty tricky. No promises.

**Comment by Darkest-Traveler** [comment]
Currently a player is doing it again right now. They put the sorry edit* Forgotten sword in the work bench, they try to make an utherium counterbalance it crashes the server with the micklus tetra string posted in the second crash log. Undergarden tools in the workbench upgraded with other undergarden mats.

**Comment by Darkest-Traveler** [comment]
If you can't replicate it and I have this issue is it possible to just remove the undergarden/tetra patch without it causing problems in a pre-existing server world, if players already have tetra/undergarden tools/weapons?

**Comment by Darkest-Traveler** [comment]
I've tried replicating it but it didn't work for me and I think its tied to the cutting a new one advancement itself, when you put forgotten sword in the table and put counter balance on it when getting cutting a new one it crashes the server and gives that report.

**Comment by Darkest-Traveler** [comment]
Latest update. The Sword is enchanted, it is a Forgotten sword, when using utherium to make fore finger ring or counterbalance it crashes the server this has happened similarly with an undergarden axe and added parts.

**Comment by Syrikal** [comment]
Okay, good news and bad news. The good news is that a) I can confirm it's safe to remove the mod- any Tetra tools made of Undergarden materials _should_ just turn into other materials, probably maple wood (idk why); and b), I can replicate your bug. Bad news is I don't know how to fix it yet, but if I have replication then I have something to work with. I'll see what I can do.

**Comment by Syrikal** [comment]
Ha! I think I have it.

The bug happens when you attempt to modify an enchanted Forgotten Metal tool in a workbench- specifically, when the game tries to turn it into a Tetra modular tool. As far as I can tell, it doesn't care how you modify it. I got the crash when I changed the sword's flimsy hilt to an oak one, for example.

I think this is because Forgotten Metal has an enchantability of 0, which means that its Tetra magic capacity is also 0. When you turn it into a Tetra tool, it tries to apply the enchantment to the Forgotten Metal module, which causes the crash somehow. I think I've fixed it by upping Forgotten Metal's magic capacity to 1.
